### ISSUE TITLE
metricstransformprocessor handle correctly the new FlagsStruct API 

### ISSUE THREAD

**Issue by bogdandrutu** [open-telemetry/opentelemetry-collector-contrib#12925]
See groupHistogramDataPoints and groupExponentialHistogramDataPoints for the `flagsValue` hack.

**Comment by TylerHelmuth** [comment]
Closing as PR was merged

**Comment by bogdandrutu** [comment]
@TylerHelmuth what I implemented is a bit of a hack :) someone needs to do the right thing. Like maybe we need a "get as int" the flags.

**Comment by abhide** [comment]
Tried looking into this issue. 
I don't see `flagsValue` in the repo (git grep yields no results).

I tried looking at the PR referenced above, `pkg/telemetryquerylanguage/contexts/tqlmetrics/metrics.go` path doesn't exist in the repo now; but after looking through git history I think commit `fd4f1da4744618b653fccd5a0c47aff146058104` by @bogdandrutu fixes the `flagsValue` hack. 

cc @dmitryax / @TylerHelmuth
